### PR TITLE
Optimize booleans when argument is negative integer

### PR DESCRIPTION
**Pull request by MaxGraey**
`bool(-x)` ==> `bool(x)`

### CODE REVIEW

File: src/passes/OptimizeInstructions.cpp [hunk 1]
```diff
@@ -982,6 +982,14 @@ struct OptimizeInstructions
         }
       }
     } else if (auto* binary = boolean->dynCast<Binary>()) {
+      if (binary->op == SubInt32) {
+        if (auto* num = binary->left->dynCast<Const>()) {
+          if (num->value.geti32() == 0) {
+            // 0 - x   ==>   x
```
Comment (kripken) [inline src/passes/OptimizeInstructions.cpp:988]: this comment looks confusing. how about `bool(0 - x)   ==>   bool(x)`?

**Comment by kripken** [comment]
Btw, I like how small this PR is - very good this way, thanks!

File: src/passes/OptimizeInstructions.cpp (hunk 1 above)
Comment (MaxGraey) [reply src/passes/OptimizeInstructions.cpp]: Sure. Done!

**Comment by kripken** [comment]
Great!

**Pull request merged**